external_server_main.py

```diff
@@ -1,66 +1,77 @@
+
+    args = parser.parse_args()
+
+    if not os.path.isfile(args.config):
+        raise FileNotFoundError(f"Config file {os.path.abspath(args.config)} not found.")
+    if args.tls:
+        missing_fields = []
+        if not args.ca:
+            missing_fields.append("ca certificate")
+        if not args.cert:
+            missing_fields.append("PEM encoded client certificate")
+        if not args.key:
+            missing_fields.append("private key to PEM encoded client certificate")
+        if missing_fields:
+            e = argparse.ArgumentError(
+                None, f"TLS requires additional parameters. The following is missing: {', '.join(missing_fields)}"
+            )
+            raise e
+    return args
 
 
 def main() -> None:
     """Main entry of external server"""
-    args = argparse_init()
     try:
-        config = load_config(args.config)
-    except InvalidConfigError as exc:
-        print(f"Invalid config: {exc}")
+        args = parsed_script_args()
+    except argparse.ArgumentError as exc:
+        eslogger.error(f"Invalid arguments. {exc}")
+        print(f"Invalid arguments. {exc}")
         sys.exit(1)
 
-    if (config.log_files_to_keep):
-            file_handler = logging.handlers.RotatingFileHandler(
-                filename=str(config.log_files_directory) + "/" + constants.LOG_FILE_NAME,
-                maxBytes=config.log_file_max_size_bytes,
-                backupCount=config.log_files_to_keep - 1
-            )
-            file_handler.setFormatter(logging.Formatter(constants.LOG_FORMAT))
-            logging.basicConfig(
-                level=logging.INFO,
-                format="%(name)s: %(message)s",
-                datefmt="[%X]",
-                handlers=[RichHandler(), file_handler],
-            )
-    else:
-        logging.basicConfig(
-        level=logging.INFO,
-        format="%(name)s: %(message)s",
-        datefmt="[%X]",
-        handlers=[RichHandler()],
-    )
-        
-    logger = logging.getLogger("Main")
-    logger.info(f"Loaded config:\n{config.get_config_dump_string()}")
+    try:
+        with open(args.config) as f:
+            config_dict = json.loads(f.read())
+            config = load_config(args.config)
+            configure_logging("External Server", config_dict)
+    except InvalidConfiguration as exc:
+        eslogger.error(f"Invalid config: {exc}")
+        print(f"Invalid config: {exc}")
+        sys.exit(1)
 
+    eslogger.info(f"Loaded config:\n{config.get_config_dump_string()}")
     server = ExternalServer(config)
     if args.tls:
-        if args.ca is None or args.cert is None or args.key is None:
-            logger.error(
-                "TLS requires ca certificate, PEM encoded client certificate and private key to this certificate"
-            )
-            sys.exit(1)
-        server.set_tls(args.ca, args.cert, args.key)
+        server.tls_set(args.ca, args.cert, args.key)
```

Comment:
```suggestion
        server.set_tls(args.ca, args.cert, args.key)
```
I think this is a better name.